File: src/de/schauderhaft/cleanCode/exercise5/InstanceClassifier.java
package de.schauderhaft.cleanCode.exercise5;

public class InstanceClassifier {

    boolean isTestInstance(String url) {
        return url.contains("grglmrpft") || url.contains("sysmrpt")
                || url.contains("gru23b12") || url.contains("gru23b13");
    }

    boolean isQaInstance(String url) {
        return url.contains("grglmrpfqs") || url.contains("gru23c12")
                || url.contains("sysglmqs") || url.contains("sysgmrpf-qs");
    }

    boolean isProductionInstance(String url) {
        return !(isTestInstance(url) || isQaInstance(url));
    }

}

File: src/de/schauderhaft/cleanCode/exercise5/InstanceClassifier.cs
namespace de.schauderhaft.cleanCode.exercise5
{
    public class InstanceClassifier
    {
        public bool IsTestInstance(string url)
        {
            return url.Contains("grglmrpfqs")
                    || url.Contains("gru23c12")
                    || url.Contains("sysglmqs")
                    || url.Contains("sysgmrpf-qs");
        }
        
        public bool IsQaInstance(string url)
        {
            return url.Contains("grglmrpft")
                    || url.Contains("sysmrpt")
                    || url.Contains("gru23b12")
                    || url.Contains("gru23b13");
        }
        
        public bool IsProductionInstance(string url)
        {
            return !(IsTestInstance(url) || IsQaInstance(url));
        }
    }
}